src/main/java/seedu/address/logic/parser/AddAssignmentCommandParser.java

```diff
@@ -42,30 +39,18 @@ public AddAssignmentCommand parse(String args) throws ParseException {
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAssignmentCommand.MESSAGE_USAGE));
         }
+
+        // Initializing compulsory fields
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Deadline deadline = ParserUtil.parseDeadline(argMultimap.getValue(PREFIX_DEADLINE).get());
         AssignmentName assignmentName = ParserUtil.parseAssignmentName(
                 argMultimap.getValue(PREFIX_ASSIGNMENT).get());
-        List<Status> statusList = ParserUtil.parseStatuses(argMultimap.getAllValues(PREFIX_STATUS));
 
-        Status submissionStatus;
-        Status gradingStatus;
-        switch (statusList.size()) {
-        case 2:
-            submissionStatus = statusList.get(0);
-            gradingStatus = statusList.get(1);
-            break;
-        case 1:
-            submissionStatus = statusList.get(0);
-            gradingStatus = Status.getDefault();
-            break;
-        case 0:
-            submissionStatus = Status.getDefault();
-            gradingStatus = Status.getDefault();
-            break;
-        default:
-            throw new ParseException(MESSAGE_EXPECTED_AT_MOST_TWO);
-        }
+        // Initializing non-compulsory fields
+        List<Status> statusList = ParserUtil.parseStatuses(argMultimap.getAllValues(PREFIX_STATUS), 2);
```

Comment:
Do you also want to let them provide one status ie submissionStatus? Cause this will not allow for that